habitat/tasks/nav/instance_image_nav_task.py

```diff
@@ -0,0 +1,246 @@
+        spec.sensor_type = habitat_sim.SensorType.COLOR
+        spec.resolution = img_params.image_dimensions
+        spec.hfov = img_params.hfov
+        spec.sensor_subtype = habitat_sim.SensorSubType.PINHOLE
+        self._sim.add_sensor(spec)
+
+        agent = self._sim.get_agent(0)
+        agent_state = agent.get_state()
+        agent.set_state(
+            AgentState(
+                position=agent_state.position,
+                rotation=agent_state.rotation,
+                sensor_states={
+                    **agent_state.sensor_states,
+                    spec.uuid: SixDOFPose(
+                        position=np.array(img_params.position),
+                        rotation=quaternion_from_coeff(img_params.rotation),
+                    ),
+                },
+            ),
+            infer_sensor_states=False,
+        )
+
+        # render observations from this sensor only
+        self._sim._sensors[spec.uuid].draw_observation()
+        img = self._sim._sensors[spec.uuid].get_observation()[:, :, :3]
+
+        # TODO: delete sensor safely, like sim.remove_sensor(uuid)
+        del self._sim._sensors[spec.uuid]
+        hsim.SensorFactory.delete_subtree_sensor(agent.scene_node, spec.uuid)
+        del agent._sensors[spec.uuid]
+        agent.agent_config.sensor_specifications = [
+            s
+            for s in agent.agent_config.sensor_specifications
+            if s.uuid != spec.uuid
+        ]
+
+        return img
+
+    def get_observation(
+        self,
+        *args: Any,
+        episode: InstanceImageGoalNavEpisode,
+        **kwargs: Any,
+    ) -> Optional[VisualObservation]:
+
+        if len(episode.goals) == 0:
+            logger.error(
+                f"No goal specified for episode {episode.episode_id}."
+            )
+            return None
+        if not isinstance(episode.goals[0], InstanceImageGoal):
+            logger.error(
+                f"First goal should be InstanceImageGoal, episode {episode.episode_id}."
+            )
+            return None
+
+        episode_uniq_id = f"{episode.scene_id} {episode.episode_id}"
+        if episode_uniq_id == self._current_episode_id:
+            return self._current_image_goal
+
+        img_params = episode.goals[0].image_goals[episode.goal_image_id]
+        self._current_image_goal = self._get_instance_image_goal(img_params)
+        self._current_episode_id = episode_uniq_id
+
+        return self._current_image_goal
+
+
+@registry.register_sensor
+class InstanceImageGoalHFOVSensor(Sensor):
+    """A sensor that returns the horizontal field of view (HFOV) in degrees
+    of the current episode's instance image goal.
+    """
+
+    cls_uuid: str = "instance_imagegoal_hfov"
+
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
+        return self.cls_uuid
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any) -> Space:
+        return spaces.Box(low=0, high=360, shape=(1,), dtype=np.int64)
```

Comment:
Observations are usually floats I believe. How do you process this observation? Does it make sense to make it a float and normalize?

Reply:
I made this a float but left it unnormalized since FOV values are typically operated on without normalization.